keyboards/tweetydabird/lotus58/rgb_full/keymaps/default/keymap.c

```diff
@@ -0,0 +1,204 @@
+            					      _______,   _______,   _______,  _______,         _______,  _______,   _______,   _______
+//                                  '----------'----------'----------'--------'       '--------'----------'----------'----------'
+),
+
+[_SYS] = LAYOUT(
+  XXXXXXX, XXXXXXX, XXXXXXX ,  XXXXXXX, XXXXXXX, XXXXXXX, _______,     _______, XXXXXXX, XXXXXXX, XXXXXXX, RGB_MOD, RGB_SAI, RGB_TOG,
+  QK_BOOT, XXXXXXX, KC_QWERTY, XXXXXXX, XXXXXXX, KC_ASTG,                       XXXXXXX, XXXXXXX, XXXXXXX, RGB_RMOD, RGB_SAD, RGB_M_P,
+  XXXXXXX, XXXXXXX, XXXXXXX,   XXXXXXX, XXXXXXX, KC_CAPS,                       XXXXXXX, XXXXXXX, XXXXXXX, RGB_HUI, RGB_VAI, RGB_M_B,
+  XXXXXXX, XXXXXXX, XXXXXXX,   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,     XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, RGB_HUD, RGB_VAD, RGB_M_R,
+								_______, _______, _______, _______,     _______, _______, _______, _______  \
+  )
+};
+
+#ifdef OLED_ENABLE
+
+static void render_logo(void) {
+    static const char PROGMEM qmk_logo[] = {
+        0x80,0x81,0x82,0x83,0x84,0x85,0x86,0x87,0x88,0x89,0x8a,0x8b,0x8c,0x8d,0x8e,0x8f,0x90,0x91,0x92,0x93,0x94,
+        0xa0,0xa1,0xa2,0xa3,0xa4,0xa5,0xa6,0xa7,0xa8,0xa9,0xaa,0xab,0xac,0xad,0xae,0xaf,0xb0,0xb1,0xb2,0xb3,0xb4,
+        0xc0,0xc1,0xc2,0xc3,0xc4,0xc5,0xc6,0xc7,0xc8,0xc9,0xca,0xcb,0xcc,0xcd,0xce,0xcf,0xd0,0xd1,0xd2,0xd3,0xd4,0
+    };
+    oled_write_P(qmk_logo, false);
+}
+
+
+static void print_status_narrow(void) {
+	// Create OLED content
+    oled_write_P(PSTR("\n"), false);
+	oled_write_P(PSTR(""), false);
+    oled_write_P(PSTR("Lotus -58-"), false);
+	oled_write_P(PSTR("\n"), false);
+
+    // Print current layer
+    oled_write_P(PSTR("Layer"), false);
+    switch (get_highest_layer(layer_state)) {
+        case _QWERTY:
+            oled_write_P(PSTR("-Base\n"), false);
+            break;
+        case _FN:
+            oled_write_P(PSTR("-Func\n"), false);
+            break;
+        case _NUM:
+            oled_write_P(PSTR("-Num \n"), false);
+            break;
+        case _SYS:
+            oled_write_P(PSTR("-Sys \n"), false);
+            break;
+        default:
+            oled_write_P(PSTR("Undef"), false);
+    }
+	
+	// Display autoshift
+    oled_write_P(PSTR("\n"), false);
+    led_t led_usb_state = host_keyboard_led_state();
+    oled_write_ln_P(PSTR("Caps- lock"), led_usb_state.caps_lock);
+	bool autoshift = get_autoshift_state();
+	oled_write_P(PSTR("\n"), false);
+	oled_write_P(PSTR("Auto-Shift"), autoshift);
+	oled_write_P(PSTR("\n"), false);
+	
+}
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+	// Rotate OLED to fit
+    if (is_keyboard_left()) {
+        return OLED_ROTATION_90;
+	}
+    return OLED_ROTATION_270;
+}
+
+bool oled_task_user(void) {
+	// Render the OLED
+    if (is_keyboard_master()) {
+        print_status_narrow();
+    } else {
+        render_logo();
+    }
+	return false;
+}
+
+#endif
```

Comment:
Could you move the oled code to the keyboard's c file, and use the kb callback instead?  This allows for a default implementation that can be overriden, and keeps the default keymap simple).

Reply:
If so, I probably need a hint or two, as I have no idea how to it seems.

Just moving the code obviously didn't work very well.